zilliqa/src/consensus.rs

```diff
@@ -353,6 +358,10 @@ impl Consensus {
             state,
             db,
             new_transactions_priority: BTreeMap::new(),
+            // Seed the rng with the node's public key
+            rng: <rand_chacha::ChaCha8Rng as rand_core::SeedableRng>::seed_from_u64(
+                BigEndian::read_u64(secret_key.node_public_key().as_bytes().as_slice()),
```

Comment:
```suggestion
                u64::from_be_bytes(secret_key.node_public_key().as_bytes()[..8].try_into().unwrap()),
```
to avoid the extra dependency?

(copied from the implementation of `BigEndian::read_u64` :stuck_out_tongue:)

Reply:
Yes great, I should really have thought of this myself...